### PR TITLE
Fix/UI workout screens

### PR DESCRIPTION
**Pull request by nadaedh**
### **Summary:**

This pull request focuses on aligning the user interface (UI) of the workout screens with the Figma designs to ensure a consistent and visually appealing experience. Specific improvements have been made to enhance the usability and visual style across multiple components, including buttons, text fields, and headers.

### **Modified Files:**
**- WorkoutCreationScreen.kt:** Updated layout to better reflect the spacing and alignment outlined in the Figma designs. Introduced reusable styles for text fields and buttons.
**- ImportOrCreateScreen.kt:** Enhanced the button designs by integrating gradient backgrounds and adding spacing consistency.
**- SessionSelectionScreen.kt:** Refined card shadows, text alignment, and container colors for improved aesthetics.
**- DateTimePicker.kt:** Replaced outdated design components with styled cards and custom text formatting.
**- TopBar.kt:** Introduced gradient backgrounds for a modernized header style.
**- Dimensions.kt** and **FontSizes.kt (new):** Centralized sizing and font definitions to eliminate hardcoding and ensure consistency.
**- strings.xml:** Added and modified string resources for better localization and labeling.

### **Future Work:**
- Extend these UI enhancements to additional screens, such as the screens to create workout plans and edit them, to ensure consistency across the application.
- Optimize UI performance by further modularizing components for reuse.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/composables/DateTimePicker.kt [hunk 1]
```diff
@@ -46,17 +72,40 @@ fun DateTimePicker(
...
+            },
+        elevation = CardDefaults.cardElevation(4.dp)) {
+          Row(
+              verticalAlignment = Alignment.CenterVertically,
+              modifier = Modifier.fillMaxSize().padding(horizontal = Dimensions.LargePadding)) {
+                Text(
+                    text = displayText,
+                    style =
+                        MaterialTheme.typography.bodyMedium.copy(
+                            color = if (selectedDateTime == null) Color.Gray else Color.Black),
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/composables/DateTimePicker.kt:100]: You can use our color in Color.kt !

(You can use NeutralGray and Black)

File: app/src/main/java/com/android/sample/ui/composables/DateTimePicker.kt [hunk 2]
```diff
@@ -46,17 +72,40 @@ fun DateTimePicker(
...
+                    modifier = Modifier.weight(1f))
+
+                Icon(
+                    imageVector = Icons.Default.DateRange,
+                    contentDescription = "Select Date",
+                    tint = Color.Gray,
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/composables/DateTimePicker.kt:106]: Same here.

File: app/src/main/java/com/android/sample/ui/composables/TopBar.kt [hunk 3]
```diff
@@ -26,16 +31,21 @@ import com.android.sample.ui.theme.DarkBlue
...
-            text = stringResource(id = title),
-            fontSize = 20.sp,
-            fontWeight = FontWeight.Bold,
-            color = Color.White,
-            modifier = Modifier.testTag("ScreenTitle"))
-      },
-      navigationIcon = { ArrowBack(navigationActions, Color.White) },
-      colors = TopAppBarDefaults.smallTopAppBarColors(containerColor = DarkBlue),
-      modifier = Modifier.testTag("TopBar"))
+  val gradientBrush = Brush.horizontalGradient(colors = listOf(TitleBlue, TopBarBlue))
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/composables/TopBar.kt:34]: Maybe you could put this in Color.kt, to be able to reuse it for others screens

File: app/src/main/java/com/android/sample/ui/workout/SessionSelectionScreen.kt [hunk 4]
```diff
@@ -70,12 +75,13 @@ fun SessionCard(session: Session, onSessionClick: (Session) -> Unit) {
           modifier = Modifier.fillMaxSize().testTag("image_${session.title}"))
       Text(
           text = session.title,
-          style = MaterialTheme.typography.bodyLarge,
+          style =
+              MaterialTheme.typography.bodyLarge.copy(
+                  fontSize = FontSizes.SubtitleFontSize,
+                  fontWeight = FontWeight.ExtraBold,
+                  shadow = Shadow(color = Color.Black, offset = Offset(4f, 4f), blurRadius = 6f)),
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/SessionSelectionScreen.kt:82]: Same

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 5]
```diff
@@ -108,74 +118,75 @@ fun WorkoutCreationScreen(
...
                         },
-                        modifier = Modifier.testTag("nextButton")) {
-                          Row(verticalAlignment = Alignment.CenterVertically) {
-                            Text("Next")
-                            Icon(
-                                imageVector = Icons.AutoMirrored.Filled.ArrowForward,
-                                contentDescription = "Next")
-                          }
+                        shape = LeafShape,
+                        colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent),
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:164]: Same

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 6]
```diff
@@ -108,74 +118,75 @@ fun WorkoutCreationScreen(
...
+                                .testTag("nextButton")) {
+                          Box(
+                              modifier =
+                                  Modifier.fillMaxSize()
+                                      .background(brush = gradientBrush, shape = LeafShape),
+                              contentAlignment = Alignment.Center) {
+                                Row(verticalAlignment = Alignment.CenterVertically) {
+                                  Text(
+                                      text = "Next",
+                                      color = Color.White,
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:180]: Same

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 7]
```diff
@@ -108,74 +118,75 @@ fun WorkoutCreationScreen(
...
+                                      fontSize = FontSizes.SubtitleFontSize,
+                                      fontWeight = FontWeight.Bold)
+                                  Icon(
+                                      imageVector = Icons.AutoMirrored.Filled.ArrowForward,
+                                      contentDescription = "Next",
+                                      tint = Color.White,
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:186]: Same

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 8]
```diff
@@ -505,3 +516,69 @@ fun LazyListScope.exerciseListItems(
...
+          Row(
+              verticalAlignment = Alignment.CenterVertically,
+              modifier = Modifier.fillMaxSize().padding(horizontal = Dimensions.LargePadding)) {
+                TextField(
+                    value = value,
+                    onValueChange = onValueChange,
+                    placeholder = {
+                      Text(
+                          text = placeholder,
+                          style = MaterialTheme.typography.bodyMedium.copy(color = Color.Gray))
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:564]: Same

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 9]
```diff
@@ -505,3 +516,69 @@ fun LazyListScope.exerciseListItems(
...
+                    },
+                    colors =
+                        TextFieldDefaults.colors(
+                            focusedContainerColor = Color.Transparent,
+                            unfocusedContainerColor = Color.Transparent,
+                            cursorColor = TitleBlue,
+                            focusedTextColor = Color.Black,
+                            unfocusedTextColor = Color.Black,
+                            focusedIndicatorColor = Color.Transparent,
+                            unfocusedIndicatorColor = Color.Transparent),
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:574]: Omg fix me that ahahah

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 10]
```diff
@@ -505,3 +516,69 @@ fun LazyListScope.exerciseListItems(
...
+                    modifier = Modifier.weight(1f))
+                Icon(
+                    imageVector = Icons.Default.Edit,
+                    contentDescription = "Edit",
+                    tint = Color.Gray,
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:579]: Same

**Review by didi2101 — CHANGES_REQUESTED** [review]
Very good Design and very well implemented, like the Figma !

Nice idea to Create Dimensions.kt.

All I can say is just correct the colors, they have to come from Color.kt

File: app/src/main/java/com/android/sample/ui/composables/DateTimePicker.kt (hunk 1 above)
Comment (nadaedh) [reply app/src/main/java/com/android/sample/ui/composables/DateTimePicker.kt]: Done

File: app/src/main/java/com/android/sample/ui/composables/TopBar.kt (hunk 3 above)
Comment (nadaedh) [reply app/src/main/java/com/android/sample/ui/composables/TopBar.kt]: Done

File: app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt [hunk 11]
```diff
@@ -561,22 +566,22 @@ fun CustomTextField(
...
+                            focusedIndicatorColor = Transparent,
+                            unfocusedIndicatorColor = Transparent),
                     modifier = Modifier.weight(1f))
                 Icon(
                     imageVector = Icons.Default.Edit,
                     contentDescription = "Edit",
-                    tint = Color.Gray,
+                    tint = NeutralGrey,
                     modifier = Modifier.size(Dimensions.iconSize))
               }
```
Comment (didi2101) [inline app/src/main/java/com/android/sample/ui/workout/WorkoutCreationScreen.kt:586]: Perfect !

**Review by didi2101 — APPROVED** [review]
Changes are done.

Approved.

**Pull request merged**